src/macaron/slsa_analyzer/checks/provenance_available_check.py

```diff
@@ -67,6 +75,246 @@ def __init__(self) -> None:
+                    build_tool=Gradle() as gradle,
+                    package_registry=JFrogMavenRegistry() as jfrog_registry,
+                ) as data_entry:
+                    group_id = gradle.get_group_id(repo_local_path)
+                    if not group_id:
+                        continue
+
+                    artifact_ids = jfrog_registry.get_artifact_ids(group_id)
+
+                    provenance_assets: list[JFrogMavenAsset] = []
+
+                    for artifact_id in artifact_ids:
+                        latest_version = jfrog_registry.get_latest_version(group_id, artifact_id)
+                        if not latest_version:
+                            continue
+
+                        provenance_assets.extend(
+                            jfrog_registry.fetch_downloadable_assets(
+                                group_id=group_id,
+                                artifact_id=artifact_id,
+                                version=latest_version,
+                                extensions=provenance_extensions,
+                            )
+                        )
+
+                    if not provenance_assets:
+                        continue
+
+                    logger.info("Found the following provenance assets:")
+                    for provenance_asset in provenance_assets:
+                        logger.info("* %s", provenance_asset.url)
+
+                    # Persist the provenance assets in the package registry data entry.
+                    data_entry.provenance_assets = provenance_assets
+
+                    self.download_provenances_from_jfrog_maven_package_registry(
+                        data_entry=data_entry,
+                        jfrog_maven_registry=jfrog_registry,
+                        provenance_assets=provenance_assets,
+                    )
+
+                    return [
+                        ProvenanceAvailableTable(
+                            asset_name=asset.name,
+                            asset_url=asset.url,
+                        )
+                        for asset in provenance_assets
+                    ]
+
+        return []
+
+    def download_provenances_from_jfrog_maven_package_registry(
+        self,
+        data_entry: PackageRegistryData,
+        provenance_assets: list[JFrogMavenAsset],
+        jfrog_maven_registry: JFrogMavenRegistry,
+    ) -> None:
+        """Download provenances from a JFrog Maven package registry."""
+        with tempfile.TemporaryDirectory() as temp_path:
+            # Note: In certain cases, Macaron can find the same provenance file in
+            # multiple different places on a package registry.
+            #
+            # We may consider de-duplicating this file, so that we do not run the same
+            # steps on the same file multiple times.
+
+            # Download the provenance
+            for prov_asset in provenance_assets:
+                if not jfrog_maven_registry.download_asset(
```

Comment:
Before downloading the asset, we should check for the size, which is returned by the API  and only download if it's smaller than the allowed size provided in `defaults.ini` file (check the GitHub release implementation).

Reply:
This download size check has been added in commit fc4f0c8d48fc19a7e81738fa6b9730c9f3045f16.